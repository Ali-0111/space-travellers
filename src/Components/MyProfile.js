const MyProfile = () => (
  <div>
    This page will be constructed shortly.
  </div>
);

export default MyProfile;
