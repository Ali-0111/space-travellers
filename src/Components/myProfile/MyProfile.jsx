import React from 'react';
import { useSelector } from 'react-redux';
import '../../styles/myProfile.css';

const MyProfile = () => {
  const stateData = useSelector((state) => (state.reservationListKey.reservationList));
  const reservedRockets = stateData.filter((rocket) => (rocket.reserved));

  const { missionsArray } = useSelector((store) => (store.missionsReducer));
  const myProfileArray = missionsArray?.filter((trueReserved) => trueReserved.reserved === true);

  return (
    <section className="reserved-missions-rockets">
      <div className="missions-list-container">
        <h2>My Missions</h2>
        <ul className="missions-list">
          {
            myProfileArray?.map((item) => (
              <li className="missions-item" key={item.mission_id} id={item.mission_id}>
                <h3>{item.mission_name}</h3>
              </li>
            ))
          }
        </ul>
      </div>
      <div className="rocket-list-container">
        <h2>My Rockets</h2>
        <ul className="rocket-list">
          {
            reservedRockets.map((rocket) => (
              <li className="rocket-item" key={rocket.id} id={rocket.id}>
                <h3>{rocket.name}</h3>
              </li>
            ))
          }
        </ul>
      </div>
    </section>
  );
};

export default MyProfile;
