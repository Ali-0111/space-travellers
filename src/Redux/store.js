import { configureStore, combineReducers } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsSlice';
import rocketSlice from './rockets/rocketsSlice';

const rootReducer = combineReducers({
  missionsReducer: missionsSlice.reducer,
  reservationListKey: rocketSlice.reducer,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
