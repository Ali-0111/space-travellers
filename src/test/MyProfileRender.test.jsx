import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import store from '../Redux/store';

import MyProfile from '../Components/myProfile/MyProfile';

describe('MyProfile rendering Validation', () => {
  test('MyProfile should render with /myProfile url', () => {
    const treeSnapShot = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/myProfile" element={<MyProfile />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(treeSnapShot).toMatchSnapshot();
  });
});
