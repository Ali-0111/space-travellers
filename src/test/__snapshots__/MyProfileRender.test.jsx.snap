// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile rendering Validation MyProfile should render with /myProfile url 1`] = `null`;
